sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -104,8 +112,13 @@ public String getId() {
         if (options == null) {
             options = new CosmosContainerRequestOptions();
         }
-        return database.getDocClientWrapper().deleteCollection(getLink(), ModelBridgeInternal.toRequestOptions(options))
-                   .map(response -> ModelBridgeInternal.createCosmosAsyncContainerResponse(response, database)).single();
+
+        if (!database.getClient().getTracerProvider().isEnabled()) {
+            return deleteInternal(options);
+        }
+
+        final CosmosContainerRequestOptions requestOptions = options;
+        return withContext(context -> deleteInternal(requestOptions, context)).subscriberContext(TracerProvider.callDepthAttributeFunc);
```

Comment:
ditto, here and in other places.

Reply:
its already after the tracer enable check